controllers/loadtest_controller.go

```diff
@@ -276,15 +276,36 @@ func (r *LoadTestReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 			}
 		}
 
-		adjustAvailabilityForDefaults := func(defaultPoolKey, defaultPoolName string) {
+		adjustAvailabilityForDefaults := func(defaultPoolKey, defaultPoolName string) bool {
 			if c, ok := missingPods.NodeCountByPool[defaultPoolKey]; ok {
```

Comment:
Nit: you may be able to test for `ok && c > 0` in one go.

I might also then test for `defaultPoolName == ""` since it has a `return` in it, and then do the other branch without the else.

Reply:
Done, definitely more elegant!